edge-hub/src/Microsoft.Azure.Devices.Routing.Core/endpoints/StoringAsyncEndpointExecutorFactory.cs

```diff
@@ -25,15 +25,15 @@ public StoringAsyncEndpointExecutorFactory(EndpointExecutorConfig config, AsyncE
 
         public Task<IEndpointExecutor> CreateAsync(Endpoint endpoint, ICheckpointer checkpointer) => this.CreateAsync(endpoint, checkpointer, this.config);
 
-        public Task<IEndpointExecutor> CreateAsync(Endpoint endpoint, ICheckpointer checkpointer, EndpointExecutorConfig endpointExecutorConfig)
+        public async Task<IEndpointExecutor> CreateAsync(Endpoint endpoint, ICheckpointer checkpointer, EndpointExecutorConfig endpointExecutorConfig)
         {
             Preconditions.CheckNotNull(endpoint, nameof(endpoint));
             Preconditions.CheckNotNull(checkpointer, nameof(checkpointer));
             Preconditions.CheckNotNull(endpointExecutorConfig, nameof(endpointExecutorConfig));
 
-            this.messageStore.AddEndpoint(endpoint.Id);
+            await this.messageStore.AddEndpoint(endpoint.Id);
```

Comment:
Was `this.messageStore.AddEndpoint` always async?

Reply:
Yes, but it was okay, coz the underlying operations were all sync..